metpy/plots/station_plot.py

```diff
@@ -260,13 +261,9 @@ def plot_barb(self, u, v, **kwargs):
         """
         kwargs = self._make_kwargs(kwargs)
 
-        # Handle transforming our center points. CartoPy doesn't like 1D barbs
-        # TODO: This can be removed for cartopy > 0.14.3
-        if hasattr(self.ax, 'projection') and 'transform' in kwargs:
-            trans = kwargs.pop('transform')
-            x, y, _ = self. ax.projection.transform_points(trans, self.x, self.y).T
-        else:
-            x, y = self.x, self.y
+        # Strip units, CartoPy transform doesn't like
```

Comment:
Any advantage/disadvantage to using `.m`? I suppose this won't error if no units are attached, but that condition should never occur anyhow.

Reply:
I'm not sure why you think barbs will never be given values without units.

Reply:
Guess we don't do a unit check here on entry do we.. units... ugh

Reply:
Because the plotting code is unit-agnostic.